Code/GraphMol/Fingerprints/FingerprintGenerator.cpp

```diff
@@ -227,9 +227,9 @@ SparseBitVect *FingerprintGenerator<OutputType>::getSparseFingerprint(
         // for every bound in the d_countBounds in dp_fingerprintArguments, set
         // a bit if the occurrence count is equal or higher than the bound for
         // that bit
-        if (val.second >= dp_fingerprintArguments->d_countBounds[i]) {
-          result->setBit(
-              val.first * dp_fingerprintArguments->d_countBounds.size() + i);
+        const auto& bounds_count = dp_fingerprintArguments->d_countBounds;
+        if (val.second >= static_cast<int>(bounds_count[i])) {
```

Comment:
This might be hiding something else.  What types are being complained about here?

I tend to use rdcast<int>(...) because this can enable out of bounds checking in debug mode.

Reply:
Here is the rdcast definition:
```
#ifdef RDDEBUG
// use rdcast to convert between types
//  when RDDEBUG is defined, this checks for
//  validity (overflow, etc)
//  when RDDEBUG is off, the cast is a no-cost
//   static_cast
#define rdcast boost::numeric_cast
#else
#define rdcast static_cast
#endif
```
I rarely turn on RDDEBUG but it has helped in the past.

Reply:
I also had my reservations on this one, but I found an identical case in the same directory:
```
AtomPairs.cpp-238-    BOOST_FOREACH (SparseIntVect<boost::int64_t>::StorageType::value_type val,
AtomPairs.cpp-239-                   sres->getNonzeroElements()) {
AtomPairs.cpp-240-      for (unsigned int i = 0; i < nBitsPerEntry; ++i) {
AtomPairs.cpp:241:        if (val.second > static_cast<int>(i))
AtomPairs.cpp-242-          res->setBit(val.first * nBitsPerEntry + i);
AtomPairs.cpp-243-      }
AtomPairs.cpp-244-
```

Reply:
Compared types are `int` vs `const unsigned int`, iirc

Reply:
Sorry, I got mixed up... not that code, but the test... for which I submitted a PR in October, and actually casts into `long`:
```
testFingerprintGenerators.cpp:239:    nz = fp1->getNonzeroElements();
testFingerprintGenerators.cpp-240-    for (const auto& it : nz) {
testFingerprintGenerators.cpp-241-      for (unsigned int i = 0; i < defaultCountBounds.size(); ++i) {
testFingerprintGenerators.cpp-242-        bool isSet = static_cast<bool>(fp2->getBit(it.first * defaultCountBounds.size() + i));
testFingerprintGenerators.cpp-243-        TEST_ASSERT(isSet == (it.second >= static_cast<long>(defaultCountBounds[i])));
testFingerprintGenerators.cpp-244-      }
```

Reply:
given what `d_countBounds` is for (counting the number of times a bit was set in a count-based fingerprint), I think we're perfectly safe with a static_cast to int.